[#59144050] Integrate CFDI gem

ulices opened the pull request:
- Add examples of certificates and keys
- Generate a valid xml file through CFDI gem

Commit:
[#59144050] Integrate CFDI gem

- Add examples of certificates and keys
- Generate a valid xml file through CFDI gem

mumoc commented on lib/facturacion_electronica.rb:
```diff
@@ -1,7 +1,60 @@
+      @bill[:impuestos]
+    end
+  end
+
```
this looks better than the original :+1:

chischaschos replied:
It is not enough.
1. Please make the class name reflect its intention. 
2. Something to thing about: 
   
   > Its interface is exposed through class methods but the methods but the class uses instances variables which suggest that it is keeping state.

> When going all class methods usually seems like you don't want to keep state (which is the reason to go kind of functional), but you are actually keeping state, so, something smells weird
1. The entry method in the interface accepts four parameters, more than two parameters makes a class hard to read, either pass in an abstraction or use a hash as parameter, it may help make things clearer and cleaner.

Please go on and read this book as soon as possible:
![image](https://f.cloud.github.com/assets/135399/1510386/017c1b04-4a7f-11e3-9ffe-da1f82930bec.png)

ulices replied:
Ok thanks, working on that :)

mumoc commented on spec/lib/facturacion_electronica_spec.rb:
```diff
@@ -1,7 +1,100 @@
+  let(:sat_certificate){ File.open './spec/docs_examples/CertificadosDemo-FM/TUMG620310R95/TUMG620310R95_1210241209S.cer' }
+  let(:sat_key){ File.open 'spec/docs_examples/CertificadosDemo-FM/TUMG620310R95/TUMG620310R95_1210241209S.key.pem' }
+  let(:pac_provider){ 'FacturacionModerna' }
+  let(:bill) do
```
couldn't you move this bill hash to another file? like a yaml, or, another class, i think you did it with another similare case

ulices replied:
Good idea, working on that :)

Commit:
Desarrollar gema para timbrar CFDi en Facturacion Moderna WIP [#59144050]

Commit:
Desarrollar gema para timbrar CFDis en FacturacionModerna [#59144050]

- Refactorizacion de pruebas
- Refactorizacion de metodo FacturacionElectronica.create_cfdi
- Integracion de la clase PacProviderFM (WIP)

Commit:
Desarrollar gema para timbrar CFDis en FacturacionModerna [#59144050]

- Configurar la peticion a FacturacionModerna acorde a las credenciales
  del usuario de FM

Commit:
Desarrollar gema para timbrar CFDi's en FacturaciónModerna [#59144050]

- Refatorizacion modulo FacturacionElectronica: mover la creacion del a
  peticion de factura a un constructor

Commit:
Refactorizacion: mover los servicios dentro del namespace services

Commit:
Update README

Commit:
Point gem to crowd respository

Commit:
Desarrollar gema para timbrar CFDis en FacturacionModerna [#59144050]

- Validacion de los parametros de peticion

Commit:
Desarrollar gema para timbrar CFDis en FacturacionModerna [#59144050]

- Refactorizando validacion de parametros de peticion
